src/main/java/com/gildedrose/AgeableRepository.java

```diff
@@ -0,0 +1,82 @@
+package com.gildedrose;
+
+import com.gildedrose.goblins_grotto.Item;
+import com.gildedrose.items.*;
+
+import java.util.*;
+//separate file which is the hashmap construction, this would be imported similar to any database connection
+
+public class AgeableRepository implements DataRepository {
+
+    public Ageable get(String id) {
+        return ItemToAgeableMapper.get(id);
+    }
+
+    public Collection<Ageable> getAll() {
+        return ItemToAgeableMapper.getAll();
+    }
+
+    public void updateAll() {
+        ArrayList<Ageable> allAgeableItems = ItemToAgeableMapper.getAll();
+        for (Ageable ageableItem : allAgeableItems) {
+            ageableItem.age();
+        }
+    }
+}
+
+class ItemToAgeableMapper {
```

Comment:
This seems like mission-critical code. Do we have tests for these?

Reply:
We ended up removing the mapper when we changed the current "DataSources" contents from <String, Item> (ID, Item) to <String, Ageable>, so no mapping was needed. This also solved an issue with storing float based qualities! Win Win!